### PR TITLE
Improvements to the React docs

### PR DESCRIPTION
**Pull request by harshil1712**

### CODE REVIEW

File: js-sdk/integrations/react/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+id: overiview
```
Comment (stepan662) [inline js-sdk/integrations/react/overview.mdx:2]: typo: overview

File: js-sdk/integrations/react/overview.mdx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+title: Tolgee with React
+sidebar_label: Overview
+description: "Get started with Tolgee's integration for React. Learn how to add localization to your React applications, including support for Next.js and React Native."
+---
+
+[React](https://react.dev/) is a library that lets you build user interfaces for your applications. To provide a localized experience to your users, you can add Tolgee to React.
```
Comment (stepan662) [inline js-sdk/integrations/react/overview.mdx:8]: I don't think it's necessary to explain what is React

File: js-sdk/integrations/react/translating.mdx [hunk 3]
```diff
@@ -1,59 +1,60 @@
...
 />
 ```
 
-### Using namespaces
+## useTranslate hook vs T-component
 
-The `useTranslate` hook can be also used for loading additional namespaces.
+The `T` component has a similar API to that of the `t` function mentioned above. However, there are some considerations when deciding which to use:
+
+- To apply translation in the compnent where you don't have access to React hooks, use the `T` component.
```
Comment (stepan662) [inline js-sdk/integrations/react/translating.mdx:52]: typo: component

File: js-sdk/integrations/react/translating.mdx [hunk 4]
```diff
@@ -1,59 +1,60 @@
...
 sidebar_label: Translating
-description: 'Learn about several ways how to translate a text using Tolgee React integration: T-component or useTranslate hook.'
+description: "Tolgee enables you to implement localization in your React application with ease. You can integrate Tolgee in your React application using Tolgee's SDK for React. Learn to impelement translation using the useTranslate hook and the T-component."
 ---
 
 import TFunctionHint from '../../shared/_TFunctionHint.mdx';
 import NamespacesHint from '../../shared/_NamespacesHint.mdx';
-import ExampleBanner from '../../shared/ExampleBanner';
 
-<ExampleBanner framework="React" />
```
Comment (stepan662) [inline js-sdk/integrations/react/translating.mdx:12]: Links to the example application are quite important, could we keep it here somewhere?

File: js-sdk/integrations/react/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+Tolgee provides an SDK for React, based on the [`@tolgee/web`](../../api/web_package/about) package. Using the SDK [@tolgee/react](https://www.npmjs.com/package/@tolgee/react), you can connect your React application to the Tolgee platform. This SDK enables you to implement [in-context translations](../../in-context), [language detection](../../language#language-detection), interpolation, and other advanced features.
+
+Follow the [installation guide](./installation) to add the Tolgee React SDK to your existing React application. You can also check the [example application](https://github.com/tolgee/react-example) to see how to integrate Tolgee with React. If you are building your application with [Next.js](https://nextjs.org/), follow our [Next.js documentation](./next/introduction) for guidance. For React Native applications, view our [React Native documentation](./react_native) to learn how to use Tolgee.
```
Comment (stepan662) [inline js-sdk/integrations/react/overview.mdx:12]: The overview is nice, however it is very dense text with not much structure. Could we maybe make it a bit more structure (e.g. using bullet points) so the user can quickly find what he needs?

Snímek obrazovky 2024-07-08 v 9 31 28

**Review by stepan662 — CHANGES_REQUESTED** [review]

File: js-sdk/integrations/react/translating.mdx (hunk 4 above)
Comment (harshil1712) [reply js-sdk/integrations/react/translating.mdx:12]: The example on the top take away the focus and feels like that is the first thing we want the users do. I have tried updating the component, and have also added it to the bottom of the pages. Let me know what you think. I am attaching the screenshots here.
Translating _ Tolgee

Installation _ Tolgee

Comment (stepan662) [reply js-sdk/integrations/react/translating.mdx:12]: Yeah, thats cool 👍

Comment (stepan662) [reply js-sdk/integrations/react/translating.mdx:12]: Just please fix the build, I think there are just some broken links, you can find out, if you run `npm run build` locally.

**Review by stepan662 — CHANGES_REQUESTED** [review]

File: js-sdk/integrations/react/tags_interpolation.mdx [hunk 6]
```diff
@@ -31,11 +42,15 @@ const Component = () => {
...
+The above example code will render the following text.
 
 ![Formatted text](/img/docs/text_formatted.png)
 
-You can also pass a function which will return React component:
+## 2. Passing a React component
+
+### Example usage
+
+You can also pass a React component:
```
[the remark this answers is not in the document]
Comment (harshil1712) [reply js-sdk/integrations/react/tags_interpolation.mdx:53]: Does this description sound correct: You can also pass a function that returns a React node.

File: js-sdk/integrations/react/tags_interpolation.mdx [hunk 7]
```diff
@@ -31,11 +42,15 @@ const Component = () => {
 };
 ```
 
-will render
+The above example code will render the following text.
 
 ![Formatted text](/img/docs/text_formatted.png)
 
-You can also pass a function which will return React component:
+## 2. Passing a React component
```
[the remark this answers is not in the document]
Comment (harshil1712) [reply js-sdk/integrations/react/tags_interpolation.mdx:49]: Thank you for the feedback. I misunderstood it. Updated the title to `Passing a function`

File: js-sdk/integrations/react/tags_interpolation.mdx (hunk 6 above)
[the remark this answers is not in the document]
Comment (stepan662) [reply js-sdk/integrations/react/tags_interpolation.mdx:53]: Yeah, I think that's more accurate

[the remark this answers is not in the document]
Comment (stepan662) [reply js-sdk/integrations/react/tags_interpolation.mdx:53]: We can metion that it recieves the content as an argument `(content: React.ReactNode) => React.ReactNode`

**Review by stepan662 — APPROVED** [review]

**Pull request merged**